index.js

```diff
@@ -242,52 +213,123 @@ const parseArgs = (config = { __proto__: null }) => {
         }
       }
       ArrayPrototypeUnshiftApply(remainingArgs, expanded);
+      groupCount = expanded.length;
       continue;
     }
 
     if (isShortOptionAndValue(arg, options)) {
       // e.g. -fFILE
       const shortOption = StringPrototypeCharAt(arg, 1);
-      const longOption = findLongOptionForShort(shortOption, options);
-      const optionValue = StringPrototypeSlice(arg, 2);
-      checkOptionUsage(longOption, optionValue, options, `-${shortOption}`, strict, allowPositionals);
-      storeOption(longOption, optionValue, options, result.values);
+      const optionName = findLongOptionForShort(shortOption, options);
+      const value = StringPrototypeSlice(arg, 2);
+      ArrayPrototypePush(
+        tokens,
+        { kind: 'option', optionName, optionUsed: `-${shortOption}`,
+          index, value, inlineValue: true });
       continue;
     }
 
     if (isLoneLongOption(arg)) {
       // e.g. '--foo'
-      const longOption = StringPrototypeSlice(arg, 2);
-      let optionValue;
-      if (optionsGetOwn(options, longOption, 'type') === 'string' &&
+      const optionName = StringPrototypeSlice(arg, 2);
+      let value;
+      let inlineValue;
+      if (optionsGetOwn(options, optionName, 'type') === 'string' &&
           isOptionValue(nextArg)) {
         // e.g. '--foo', 'bar'
-        optionValue = ArrayPrototypeShift(remainingArgs);
-        checkOptionLikeValue(longOption, optionValue, arg, strict);
+        value = ArrayPrototypeShift(remainingArgs);
+        inlineValue = false;
       }
-      checkOptionUsage(longOption, optionValue, options,
-                       arg, strict, allowPositionals);
-      storeOption(longOption, optionValue, options, result.values);
+      ArrayPrototypePush(
+        tokens,
+        { kind: 'option', optionName, optionUsed: arg,
```

Comment:
Do we need the `option` prefix for `optionName`? Would `token.name` be sufficient? I can't think of another case for named arguments unless we wanted to add a distinction for `long` and `short` option names. Even if that were the case, it might be better to describe them as distinct `kind`s (e.g. `kind: 'LongOption'` and `kind: 'ShortOption'`) where authors can evaluate a combination of `kind` and `name`. Note: I'm not suggesting we separate option `kind`s. I like what you have (just thinking out loud) 👍 

Here are some examples of `node.name` in practice:
<details>
  <summary>acorn</summary>

![image](https://user-images.githubusercontent.com/32409546/171317094-ba0194f8-1a62-4cd7-950b-56ad03b3b6e9.png)
</details>

<details>
  <summary>postcss</summary>

![image](https://user-images.githubusercontent.com/32409546/171317260-c8e804d4-d834-444d-9764-3ec111b28eba.png)
</details>


Also, I quite liked the `longOption`/`shortOption` naming convention we were using throughout the code/utilities and this update is nudging a global change. That being said, I'm always open to change if folks prefer this convention 🙂

Reply:
`optionName/name`: you are right `name` probably sufficient, and we are leaning towards short names, I'll give it a try.

>  I quite liked the `longOption/shortOption` naming convention

Me too, it helped clarify the code for me at the time. I might have gone too far with using the token property names for variables. I'll try and keep that in mind when I look at `optionName/name`.

Reply:
Changed to `name`, restored `longOption`